### PR TITLE
Adding correct state transitions and fixing couple of typos

### PR DESCRIPTION
**Pull request by ankitdixit**

### CODE REVIEW

File: logdevice/common/configuration/nodes/ShardStateTracker.cpp [hunk 1]
```diff
@@ -19,16 +19,20 @@ namespace {
 ShardState::Update
 transitionToNonIntermediaryState(const ShardState& shard_state) {
   ShardState::Update update{};
-  update.transition = StorageStateTransition::OVERRIDE_STATE;
-  update.conditions = Condition::FORCE;
-
-  ShardState::Update::StateOverride state_override{};
-  state_override.storage_state =
-      toNonIntermediaryState(shard_state.storage_state);
```
Comment (MohamedBassem) [inline logdevice/common/configuration/nodes/ShardStateTracker.cpp:27]: I don't think this `toNonIntermediaryState` state function is used in anything useful anymore, right? If so, let's remove it.

File: logdevice/common/configuration/nodes/ShardStateTracker.cpp [hunk 2]
```diff
@@ -19,16 +19,20 @@ namespace {
...
-  state_override.metadata_state =
-      toNonIntermediaryState(shard_state.metadata_state);
-  // TODO: handle UNRECOVERABLE flag
-  update.state_override = std::move(state_override);
+
+  if (shard_state.storage_state == StorageState::NONE_TO_RO) {
+    update.transition = StorageStateTransition::COMMIT_READ_ENABLED;
+    update.conditions = Condition::COPYSET_CONFIRMATION;
+  }
+  else if (shard_state.storage_state == StorageState::RW_TO_RO) {
```
Comment (MohamedBassem) [inline logdevice/common/configuration/nodes/ShardStateTracker.cpp:27]: Can you run `clang-format` on the changes? :)

File: logdevice/common/configuration/nodes/ShardStateTracker.cpp (hunk 1 above)
Comment (ankitdixit) [reply logdevice/common/configuration/nodes/ShardStateTracker.cpp:27]: done

File: logdevice/common/configuration/nodes/ShardStateTracker.cpp (hunk 2 above)
Comment (ankitdixit) [reply logdevice/common/configuration/nodes/ShardStateTracker.cpp]: done

**Pull request closed without merging**